src/main/java/uk/gov/hmcts/reform/sendletter/health/ServiceBusHealthIndicator.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.reform.sendletter.health;
+
+import com.microsoft.azure.servicebus.IQueueClient;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+import java.util.function.Supplier;
+
+@Component
+public class ServiceBusHealthIndicator implements HealthIndicator {
+
+    private final Supplier<IQueueClient> queueClientSupplier;
+
+    public ServiceBusHealthIndicator(Supplier<IQueueClient> queueClientSupplier) {
+        this.queueClientSupplier = queueClientSupplier;
+    }
+
+    @Override
+    public Health health() {
+        try {
+            queueClientSupplier.get().close();
+            return Health.up().build();
+        } catch (Exception exc) {
+            return Health.down().withException(exc).build();
+        }
```

Comment:
Is there a reason why abstract implementation is dismissed over interface? it will just make

```java
doHealthCheck() {
    queue.get.close;
    builder.up();
}
```

Reply:
I find a method that returns void and modifies passed parameter obscure

Reply:
I'm also a fan of the explicit version, but then all health indicators need to be implemented in a consistent way.

Reply:
@lgonczar this is the only healthcheck in this project atm.

Most healthchecks in consumer implement `HealthIndicator`

Reply:
Oh, sorry